src/software/ai/hl/stp/tactic/shadow_enemy/shadow_enemy_fsm.h

```diff
@@ -0,0 +1,219 @@
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/geom/algorithms/distance.h"
+
+struct ShadowEnemyFSM
+{
+    class BlockShotState;
+    class BlockPassState;
+    class StealAndChipState;
+
+    // this struct defines the unique control parameters that the ShadowEnemyFSM requires
+    // in its update
+    struct ControlParams
+    {
+        // The Enemy Threat indicating which enemy to shadow
+        std::optional<EnemyThreat> enemy_threat;
+
+        // How far from the enemy the robot will position itself to shadow
+        double shadow_distance;
+    };
+
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+
+    // Distance to chip the ball when trying to yeet it
+    // TODO (#1878): Replace this with a more intelligent chip distance system
+    static constexpr double YEET_CHIP_DISTANCE_METERS = 2.0;
+
+
+    /**
+     * Calculates the point to block the pass to the robot we are shadowing
+     *
+     * @param ball_position The position of the ball
+     * @param shadowee The enemy robot we are shadowing
+     * @param shadow_distance The distance our friendly robot will position itself away
+     * from the shadowee
+     */
+    static Point findBlockPassPoint(const Point &ball_position, const Robot &shadowee,
+                                    const double &shadow_distance)
+    {
+        Vector enemy_to_shadowee_vector = ball_position - shadowee.position();
+
+        return shadowee.position() + enemy_to_shadowee_vector.normalize(shadow_distance);
+    }
+
+
+    /**
+     * Calculates the point to block the shot from the robot we are shadowing
+     *
+     * @param robot The robot that is shadowing
+     * @param field The field to shadow on
+     * @param friendlyTeam The friendly team
+     * @param enemyTeam The enemy team
+     * @param shadowee The enemy robot we are shadowing
+     * @param shadow_distance The distance our friendly robot will position itself away
+     * from the shadowee
+     */
+    static Point findBlockShotPoint(const Robot &robot, const Field &field,
+                                    const Team &friendlyTeam, const Team &enemyTeam,
+                                    const Robot &shadowee, const double &shadow_distance)
+    {
+        std::vector<Robot> robots_to_ignore = {robot};
+        if (friendlyTeam.goalie().has_value())
+        {
+            robots_to_ignore.emplace_back(friendlyTeam.goalie().value());
+        }
+
+        auto best_enemy_shot_opt =
+            calcBestShotOnGoal(field, friendlyTeam, enemyTeam, shadowee.position(),
+                               TeamType::FRIENDLY, robots_to_ignore);
+
+        Vector enemy_shot_vector = Vector(0, 0);
+        if (best_enemy_shot_opt)
+        {
+            enemy_shot_vector =
+                best_enemy_shot_opt->getPointToShootAt() - shadowee.position();
+        }
+        else
+        {
+            enemy_shot_vector = field.friendlyGoalCenter() - shadowee.position();
+        }
```

Comment:
nit: this can be simplified a bit
```suggestion
        Vector enemy_shot_vector =  field.friendlyGoalCenter() - shadowee.position();
        if (best_enemy_shot_opt)
        {
            enemy_shot_vector =
                best_enemy_shot_opt->getPointToShootAt() - shadowee.position();
        }
```

Reply:
Fixed